packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -59,7 +60,7 @@ export interface DecorateQueryKeyable {
    *
    * @see https://tanstack.com/query/latest/docs/framework/react/guides/filters
    */
-  queryFilter: () => QueryFilters;
+  queryFilter: (input?: undefined, filters?: QueryFilters) => QueryFilters;
```

Comment:
I'm unsure about this compromise, it seems the only option given we can't know what's a leaf or route node at runtime, but I don't like having to pass undefined to get to the overrides